File: app/src/main/java/br/com/yourapp/pokeapi/models/PokemonSearchResponse.java
package br.com.yourapp.pokeapi.models;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class PokemonSearchResponse implements Serializable {
    @SerializedName("count")
    public int photos;

    @SerializedName("next")
    public String next;

    @SerializedName("previous")
    public String previous;

    @SerializedName("results")
    public List<Pokemon> results;

    public String toJSON() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }
}

File: app/src/main/java/br/com/yourapp/pokeapi/models/PokemonSearchResponse.kt
package br.com.yourapp.pokeapi.models

import com.google.gson.Gson
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class PokemonSearchResponse : Serializable {
    @SerializedName("count")
    var photos = 0

    @SerializedName("next")
    var next: String? = null

    @SerializedName("previous")
    var previous: String? = null

    @SerializedName("results")
    var results: List<Pokemon>? = null
    fun toJSON(): String {
        val gson = Gson()
        return gson.toJson(this)
    }
}